### PR TITLE
Feature :  Search Todo

### PR DESCRIPTION
**Pull request by ich-wtag**
**Overview**

[Task-5: Search Todo](https://trello.com/c/4wrjnUOP/5-task-5-search-todo)
This PR includes **Todo Search** functionality. Here is an input button with a **Search button**. User can search the desired todo items in the list by using **Todo's Title** . If the input box is empty and user click's the Search Button, then whole Todo's 
 
**Types of Changes**

- [x] There is a new input element on the top of document with a button along with it.
- [x] When a search phrase is entered and search button is clicked, relevant todos are shown. 

Review
Please have a close look on this PR and kindly put your valuable comments.

[search_todo.webm](https://github.com/ich-wtag/Todo-App_Vanilla_Js/assets/153156568/32f1f613-79a8-4721-8a9c-21f1acc4827d)

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -22,6 +24,7 @@ const addTodoHandler = () => {
         id: new Date().getTime(),
         title: todoTitle,
         isEditing: false,
+        isCompleted: false,
```
Comment (hhu-wtag) [inline script.js:31]: This change is not supposed to be in this PR.

File: script.js [hunk 2]
```diff
@@ -109,7 +111,15 @@ const markDoneTodoHandler = (
     cancelButton.classList.add("hide");
 
     todo.title = inputElement.value;
-    todo.isComplete = true;
+    todo.isCompleted = true;
+};
+
+const searchHandler = () => {
+    let searchedArray = todos.filter((todo) =>
```
Comment (hhu-wtag) [inline script.js]:
```suggestion
    const searchedArray = todos.filter((todo) =>
```

File: script.js [hunk 3]
```diff
@@ -132,6 +142,12 @@ const createTodoElement = (todo) => {
     $cancelButton.classList.add("hide");
     $inputElement.classList.add("hide");
 
+    if (todo.isCompleted) {
+        $paragraphElement.classList.add("done-todo");
+        $doneButton.classList.add("hide");
+        $editButton.classList.add("hide");
+    }
```
Comment (hhu-wtag) [inline script.js]: ditto

File: script.js [hunk 4]
```diff
@@ -112,6 +115,16 @@ const markDoneTodoHandler = (
     todo.isComplete = true;
 };
 
+const searchHandler = () => {
+    let searchedValue = $searchInput.value.toLowerCase().trim();
```
Comment (nmi-wtag) [inline script.js]:
```suggestion
    const searchedValue = $searchInput.value.toLowerCase().trim();
```

File: script.js [hunk 5]
```diff
@@ -185,3 +204,4 @@ const renderTodos = (todos) => {
 };
 
 $addButton.addEventListener("click", addTodoHandler);
+$searchButton.addEventListener("click", searchHandler);
```
Comment (nmi-wtag) [inline script.js]:
```suggestion
$searchInput.addEventListener("keyup", searchHandler);
```

**Review by nmi-wtag — APPROVED** [review]

**Review by nmi-wtag — APPROVED** [review]

**Review by hhu-wtag — APPROVED** [review]
Approved

**Pull request merged**